### ISSUE TITLE
CPUs load only on master node and no CPUs load on workers nodes during evaluation of the model

### ISSUE THREAD

**Issue by AliZuhair** [maxpumperla/elephas#89]
I've installed Spark on Raspberry PI (3 nodes) cluster. I am using Keras to test (evaluate) a model that is previously trained. The problem is that there is no improvement when I run the code on the Spark. I am getting the same processing time when I run the code on single raspberry Pi and on the Spark cluster. After monitoring the CPUs load, it turned out there is only CPU load on master node, and zero load on the workers node. 

    import findspark
    findspark.init()
    import numpy as np
    from keras.optimizers import SGD
    from elephas import optimizers as elephas_optimizers
    from keras.datasets import cifar10
    from keras.utils import np_utils
    from keras.models import load_model
    from keras.utils import np_utils

    from elephas.spark_model import SparkModel
    from elephas.utils.rdd_utils import to_simple_rdd
    from elephas import optimizers as elephas_optimizers

    from pyspark import SparkContext, SparkConf
    from keras import backend as K
    K.set_image_dim_ordering('th')

    # load data
    (X_train, y_train), (X_test, y_test) = cifar10.load_data()

    # normalize inputs from 0-255 to 0.0-1.0
    X_test = X_test.astype(np.float32)
    np.divide(X_test,255.0, out = X_test)

    # one hot encode outputs
    y_test = np_utils.to_categorical(y_test)

    model = load_model('/home/pi/Desktop/my_model.h5')

    conf = SparkConf()
    conf.set("spark.app.name", 'test')
    conf.set("spark.master", "spark://node1:7077")
    conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer") 

    sc = SparkContext(conf=conf)
    adagrad = elephas_optimizers.Adagrad()
    sgd = SGD(lr=0.1)
    spark_model = SparkModel(sc,
                             model,
                             optimizer=adagrad,
                             frequency='epoch',
                             mode='asynchronous',
                             num_workers=2,master_optimizer=sgd)

    score = spark_model.master_network.evaluate(X_test, y_test, verbose=0)
    print('Test loss:', score[0])
    print('Test accuracy:', score[1])`

**Comment by maxpumperla** [comment]
@AliZuhair weird. is this also true in `synchronous` mode? sync uses only built-in spark functionality, no hand-rolled parameter services. if that's the case there's something wrong with the HTTP param server.

In any case I'd encourage you to try latest elephas master, ton of bug fixes etc. might affect your issue as well.

**Comment by danielenricocahall** [comment]
Is this still an isue?

**Comment by danielenricocahall** [comment]
Ah I just read through this again - yes, this is expected actually, as distributed inference/evaluation wasn't introduced until 1.2.0 (with a minor bug fix to evaluation in 1.2.1). If you update to that version, then do `spark_model.evaluate(...)` instead of `spark_model.master_network.evaluate(...)`, it should do distributed evaluation on the data.
